### ISSUE TITLE
`not-found.tsx` inside group does not rendered for unmatched URLs

### ISSUE THREAD

**Issue by viniciusbitt** [vercel/next.js#54980]
### Link to the code that reproduces this issue or a replay of the bug

https://codesandbox.io/p/sandbox/sleepy-chihiro-3n4tly

### To Reproduce

1. Start application in development or in production (next dev or next start)
2. Navigate to any unmatched URL
3. See that the `not-found.tsx` inside the group is not rendered

https://github.com/vercel/next.js/assets/34211799/607ada1d-281c-414d-9221-99c3ac7325f0

### Current vs. Expected behavior

Expected `not-found.tsx` to be rendered within the group when it's unmatched URL, but it only works for `notFound()` function.

### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: darwin
      Arch: x64
      Version: Darwin Kernel Version 21.6.0: Thu Jul  6 22:18:26 PDT 2023; root:xnu-8020.240.18.702.13~1/RELEASE_X86_64
    Binaries:
      Node: 18.17.0
      npm: 9.8.1
      Yarn: 1.22.19
      pnpm: N/A
    Relevant Packages:
      next: 13.4.20-canary.16
      eslint-config-next: 13.4.4
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 5.1.3
    Next.js Config:
      output: N/A
```

### Which area(s) are affected? (Select all that apply)

App Router, Routing (next/router, next/navigation, next/link)

<sub>[NEXT-1602](https://linear.app/vercel/issue/NEXT-1602/not-foundtsx-inside-group-does-not-rendered-for-unmatched-urls)</sub>

**Comment by devjiwonchoi** [comment]
This is intended--no I would say limited. The non-root `not-found` is confined to predefined URLs, and the others are caught on the root not-found.

> Good to know: In addition to catching expected notFound() errors, the root app/not-found.js file also handles any unmatched URLs for your whole application. This means users that visit a URL that is not handled by your app will be shown the UI exported by the app/not-found.js file.

I've opened a feature request for this issue.

**Comment by viniciusbitt** [comment]
I've been thinking about this being intentional, if it is, it will bring a lot of problems in certain future projects, as well as mine.

My case makes this a problem, I have a group called `(e-commerce)` and a group called `(admin)/admin`, both will have different `layout.tsx` , if I navigate to an unmatched URL within these groups, it renders root `not-found.tsx` without the group layout.

I think root `not-found.tsx` for unmatched URLs should be optional.

**Comment by viniciusbitt** [comment]
Yes, I created an issue for this PR, but I thought that the unmatched URLs would also be rendered with the proper route group layout, which is not the actual case using the `canary` version.

**Comment by viniciusbitt** [comment]
i know that in the documentation it is written that it is intentional--or not, but this raises an issue for me as described:

> My case makes this a problem, I have a group called `(e-commerce)` and a group called `(admin)/admin`, both will have different `layout.tsx` , if I navigate to an unmatched URL within these groups, it renders root `not-found.tsx` without the group layout.

**Comment by devjiwonchoi** [comment]
My bad. I misunderstood your issue. Your `(group)` is inside `group-dir` which makes your group not a root, so the not-found won't be treated as root.

**Comment by devjiwonchoi** [comment]
@viniciusbitt So it seems like this hasn't been released to [canary](https://github.com/vercel/next.js/deployments/release-canary) yet.

**Comment by viniciusbitt** [comment]
@devjiwonchoi [v13.4.20-canary.16](https://github.com/vercel/next.js/releases/tag/v13.4.20-canary.16) is in this version, and in fact it was fixed as in the example video.

**Comment by devjiwonchoi** [comment]
I see, but in your repro, it is correct to show default 404.

> My bad. I misunderstood your issue. Your (group) is inside group-dir which makes your group not a root, so the not-found won't be treated as root.

So basically what you wanted was `(group)/not-found` to be treated as root not-found right?
But #54931 was to fix `/not-found` not being rendered on notFounds inside `/(group)` - #54892 only.
I'll work on your case right now.

**Comment by devjiwonchoi** [comment]
Fully understood. I'll work on it, thank you!

**Comment by devjiwonchoi** [comment]
Hey @viniciusbitt just a reminder that your issue is expected since:

> Your (group) is inside group-dir which makes your group not a root, so the not-found inside won't be treated as root.

@huozhi I was investigating multi-root layout group routes with not-founds inside, since they don't have `app/layout`, they can't have `not-found` on the `app/`. I'm trying to clarify how to handle the unmatched URLs if global not-found is not allowed in group routes with multi-root layouts.

Here's a repro: https://codesandbox.io/p/sandbox/vibrant-bouman-8gf23v

IMO the expected behavior for `/404` is displaying the not-found inside the group which has `app/(group)/page.tsx`.

**Comment by tecoad** [comment]
@devjiwonchoi 
I am keeping up with this topic. 

I little of context: i have a (locale) group. On latest stable the custom not-found.ts at root (/app) fails to build:

`> Export encountered errors on following paths:
        /_not-found`
        
        
Even though I can see the page in during development.
If i delete the custom not-found.ts, the build works.

On the latest canary (v13.4.20-canary.23) after a fix on #54931 by @huozhi I get exactly the same build error regardless if there is a custom not-found.tsx at the root or not.

Which means that it ends up that we cannot have a custom not-found and groups in production right not.
Someone on the same issue?

**Comment by devjiwonchoi** [comment]
Please see https://github.com/vercel/next.js/issues/55191#issuecomment-1985794575

**Comment by Chukwu3meka** [comment]
This worked for me. Please see [#59180](https://github.com/vercel/next.js/issues/59180#issuecomment-2828177089)

**Comment by marcoshernanz** [comment]
The current behavior matches the documented distinction between nested `not-found.tsx` boundaries and unmatched routes. However, `global-not-found.tsx` does not cover the request for a group-specific 404 that preserves the group layout. Keeping this open as a feature request so the routing and precedence semantics can be decided explicitly.
